src/main/java/it/pagopa/pn/deliverypush/legalfacts/LegalFactGenerator.java

```diff
@@ -283,5 +314,21 @@ public String generateNotificationAARForSMS(NotificationInt notification) {
     public int getNumberOfPages( byte[] pdfBytes ) {
         return documentComposition.getNumberOfPageFromPdfBytes( pdfBytes );
     }
+
+
+    private String getAccessUrl(String iun) {
+        return String.format(pnDeliveryPushConfigs.getWebapp().getDirectAccessUrlTemplate(), iun);
+    }
+
+    private String getAccessUrlLabel() {
+        try {
+            return new URL(pnDeliveryPushConfigs.getWebapp().getDirectAccessUrlTemplate()).getHost();
+        } catch (MalformedURLException e) {
+            log.warn("cannot get host", e);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This code contains a potential information leakage in the handling of the following *exception logging*: (MalformedURLException e). Logging an exception object typically results in printing the full stack trace, which reveals variable names, method names, line numbers, and other parts of the code.

**Fix**
Release data more selectively, or declassify the data before it&#39;s released. Alternatively, switch to the DEBUG (debug() API) or TRACE (trace() API) logging levels.

**More info**
[Learn more about information leakage](https://cwe.mitre.org/data/definitions/497.html) (external link).